torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -29,23 +31,22 @@ namespace {
 
 std::string stringSlice(
     std::string string,
-    int64_t start,
-    int64_t end,
+    c10::optional<int64_t> start,
+    c10::optional<int64_t> end,
     int64_t step) {
-  TORCH_CHECK(step == 1, "Slicing a string only supports step=1");
-
-  const int64_t size = string.size();
+  int64_t start_val = start.has_value() ? start.value() : INT64_MAX;
```

Comment:
Why does this default to `INT64_MAX`? Why not just set it to 0? cc @eellison this is impeding SymInt-ification

Reply:
Ugh, I forget, but I think that was mirroring how the python slice handling code went where `None` would be encoded as `INT64_MAX`